### PR TITLE
Split server cleanup in multiple files

### PR DESCRIPTION
**Pull request by Tcharl**
This pr is meant to help the future split of server variant into multiple generator or sub folders
---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] The JDL part is updated if necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by DanielFran** [comment]
@Tcharl maybe rename method cleanupOldServerFiles to cleanupOldFiles?

**Review by DanielFran — APPROVED** [review]

**Comment by Tcharl** [comment]
@DanielFran 
As I'm not sure about the future split I prefer to keep it like this.
i.e., the future modules split could contain only one `cleanup.js` file that will clean server, client and platform files for old versions. In this case we'll be happy to have a `cleanupoldserverfile` method instead of a one fits all I guess

File: generators/server/cleanup-reactive.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/**
```
Comment (mshima) [inline generators/server/cleanup-reactive.js:1]: If we split reactive, we may need to split every other tech into imperative/reactive.
We probably should keep it in server.

**Comment by Tcharl** [comment]
Clear, I'll take it into account in the next PR (I've a 3 hours flight so will be fixed tonight)

File: generators/server/cleanup-angular.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/**
```
[the remark this answers is not in the document]
Comment (Tcharl) [reply generators/server/cleanup-angular.js:1]: Not so sure for this one: in the future this method will be in the cleanup file of angular

File: generators/server/cleanup-reactive.js (hunk 1 above)
Comment (Tcharl) [reply generators/server/cleanup-reactive.js:1]: I suggest to have a `clean-reactive` file for everything that is common, but use `if(reactive)`in the respective `cleanup-<tech>` for everything related to a particular technology (sql, angular, ...)

**Pull request merged**